nntrainer/src/addition_layer.cpp

```diff
@@ -38,8 +38,7 @@ int AdditionLayer::initialize() {
 }
 
 sharedConstTensor AdditionLayer::forwarding(sharedConstTensor in) {
-  hidden = Tensor(in->batch(), output_dim.channel(), output_dim.height(),
-                  output_dim.width());
+  hidden = Tensor(input_dim);
```

Comment:
is it right? for the forwarding during inference? Cause inference input batch_size is not batch_size as in configuration file. And is it same dim  input and hidden dimension for addition layer?

Reply:
For forwarding inference, batch_size is reset with the inference method before calling forwarding on the layers (line 380 in neuralnetwork.cpp).
The updation of batch_size in the inference method updated the input dimension and hidden dimension (output dimension) both.

Reply:
> And is it same dim input and hidden dimension for addition layer?

As far as I know, most of the layers don't change batch size. If they do in future , then their `setBatchSize()` will have be overloaded to handle that.

Reply:
I recommend to run Applications and check accuracy as well.